### PR TITLE
change snyk-severity-threshold and update README

### PR DESCRIPTION
**Pull request by FatmaTosun**
## What does this change?

change snyk-severity-threshold

## Why?

We want to be notified for `criticial` vulnerabilities

### CODE REVIEW

**Review by nhoffmann — COMMENTED** [review]
Looks good to me. But I have a question about all the secrets it needs.

**Comment by FatmaTosun** [comment]
> Looks good to me. But I have a question about all the secrets it needs.
> 
> I bought it up in the previous, now closed PR as well: [#28 (review)](https://github.com/spring-media/la-shared-github-workflows/pull/28#pullrequestreview-1445970081) Do we really need to install all these dependencies and configure all the things?

Yes, we need to 

> Looks good to me. But I have a question about all the secrets it needs.
> 
> I bought it up in the previous, now closed PR as well: [#28 (review)](https://github.com/spring-media/la-shared-github-workflows/pull/28#pullrequestreview-1445970081) Do we really need to install all these dependencies and configure all the things?

Yes we have to. I tried without install dependencies it's not working.
Screenshot 2023-06-01 at 15 40 15

When we run `snyk test  --severity-threshold=critical` it check go.mod file and list all dependencies and go through on all packages. We have private repos so that's why we need to configure.

**Comment by nhoffmann** [comment]
And what about configuring access to AWS? Do we really need that to install the dependencies?

Did you test it for the js action as well? Do we really need to install the dependencies there?

**Comment by FatmaTosun** [comment]
> And what about configuring access to AWS? Do we really need that to install the dependencies?
> 
> Did you test it for the js action as well? Do we really need to install the dependencies there?

I run & tested for js project (for interactive-admin-web) as well, but not sure we really need to set up node and pull dependencies. I thought it targets package.json file therefore we have to configure. But I apparently I was wrong, npm package manager targets package-lock.json. We already have hash of repos so it makes sense no need to set up and pull.

Screenshot 2023-06-01 at 17 36 16


Besides, I checked if there is a way to point go.sum as target file for golang projects so we wouldn't need to run extra steps. There is such option for python but not for golang.

File: README.md [hunk 1]
```diff
@@ -236,7 +264,7 @@ jobs:
       ACCESS_KEY_ID: ${{ secrets.access_key_id }}
       ACCESS_KEY_SECRET: ${{ secrets.access_key_secret }}
       LA_TECH_USER_AUTH_TOKEN: ${{ secrets.LA_TECH_USER_AUTH_TOKEN }}
-      LA_TECH_USER_SSH_KEY: ${{ secrets.la_tech_user_ssh_key }}
+      LA_TECH_USER_SSH_KEY: ${{ secrets.LA_TECH_USER_SSH_KEY }}
```
[the remark this answers is not in the document]
Comment (nhoffmann) [reply README.md:261]: Ah, this is just an improvement to the documentation of the format-unit-tests workflow and not related to the actual changes in this PR.

File: README.md [hunk 2]
```diff
@@ -197,32 +192,59 @@ jobs:
...
+jobs:
+  …
+  security:
+    uses: spring-media/la-shared-github-workflows/.github/workflows/reusable-workflow__golang__security.yml@v1
+    with:
+      snyk-severity-threshold: <vulnerabilities threshold not required and default value is critical>
+      go-version: <the go version of the project>
+    secrets:
+      ACCESS_KEY_ID: ${{ secrets.access_key_id }}
+      ACCESS_KEY_SECRET: ${{ secrets.access_key_secret }}
```
Comment (nhoffmann) [inline README.md:226]: Did you also check whether we need to set up AWS for pulling dependencies in golang? Or do we need it for something else?

File: .github/workflows/reusable-workflow__js__security.yml [hunk 3]
```diff
@@ -24,31 +19,12 @@ jobs:
...
-        run: npm install --ignore-scripts
-        env:
-          NODE_AUTH_TOKEN: ${{secrets.NPM_AUTH_TOKEN}}
-
-      - name: Install Snyk
-        uses: snyk/actions/setup@master
-      - name: Run synk test
-        run: snyk test --severity-threshold=${{ inputs.severity-threshold }}
+      - name: Run Snyk to check for vulnerabilities
+        uses: snyk/actions/node@master
```
Comment (nhoffmann) [inline .github/workflows/reusable-workflow__js__security.yml:23]: Can this reference version 0.4.0 as well, instead of `master`?

**Review by nhoffmann — COMMENTED** [review]
Looks good. Just a minor consistency thing. 
Are you planning on adding a comment to the golang action explaining why it needs to "environment to pull dependencies from non public spring-media repository"?

File: README.md (hunk 2 above)
Comment (FatmaTosun) [reply README.md]: Honestly, I was thinking if we can not restore go modules from cache we might need but I didn't check that. Now I could run without aws setup. I removed this step.

**Comment by FatmaTosun** [comment]
> Looks good. Just a minor consistency thing.
> Are you planning on adding a comment to the golang action explaining why it needs to "environment to pull dependencies from non public spring-media repository"?

Already added here https://github.com/spring-media/la-shared-github-workflows/pull/31/files#diff-ce7d3ebd258da50b2f9fec227335cb13fe19b754d9973f91c936a5209fc861ee

**Comment by nhoffmann** [comment]
> > Looks good. Just a minor consistency thing.
> > Are you planning on adding a comment to the golang action explaining why it needs to "environment to pull dependencies from non public spring-media repository"?
> 
> Already added here https://github.com/spring-media/la-shared-github-workflows/pull/31/files#diff-ce7d3ebd258da50b2f9fec227335cb13fe19b754d9973f91c936a5209fc861ee

Hey, this is not fair, you committed this after my review 🙂

**Review by nhoffmann — APPROVED** [review]
Looks good to me now.

File: README.md (hunk 2 above)
Comment (nhoffmann) [reply README.md]: Thanks for checking.

**Comment by FatmaTosun** [comment]
> > > Looks good. Just a minor consistency thing.
> > > Are you planning on adding a comment to the golang action explaining why it needs to "environment to pull dependencies from non public spring-media repository"?
> > 
> > 
> > Already added here https://github.com/spring-media/la-shared-github-workflows/pull/31/files#diff-ce7d3ebd258da50b2f9fec227335cb13fe19b754d9973f91c936a5209fc861ee
> 
> Hey, this is not fair, you committed this after my review 🙂

sorry I saw comment after commit :)

**Pull request merged**